_data-prepper/core-apis.md

```diff
@@ -0,0 +1,85 @@
+| ```POST /shutdown``` | Starts a graceful shutdown of Data Prepper. |
+| ```GET /metrics/prometheus```<br>```POST /metrics/prometheus``` | Returns a scrape of Data Prepper metrics in Prometheus text format. This API is available as a `metricsRegistries` parameter in the Data Prepper configuration file `data-prepper-config.yaml` and contains `Prometheus` as part of the registry.
+| ```GET /metrics/sys```<br>```POST /metrics/sys``` | Returns JVM metrics in Prometheus text format. This API is available as a `metricsRegistries` parameter in the Data Prepper configuration file `data-prepper-config.yaml` and contains `Prometheus` as part of the registry.
+
+## Configuring the server
+
+You can configure your Data Prepper core APIs through the `data-prepper-config.yaml` file. 
+
+### SSL/TLS connection
+
+Many of the Getting Started guides for this project disable SSL on the endpoint:
+
+```yaml
+ssl: false
+```
+
+To enable SSL on your Data Prepper endpoint, configure your `data-prepper-config.yaml` file with the following options:
+
+```yaml
+ssl: true
+keyStoreFilePath: "/usr/share/data-prepper/keystore.p12"
+keyStorePassword: "secret"
+privateKeyPassword: "secret"
+```
+
+For more information about configuring your Data Prepper server with SSL, see [Server Configuration](https://github.com/opensearch-project/data-prepper/blob/main/docs/configuration.md#server-configuration). If you are using a self-signed certificate, you can add the `-k` flag to quickly test sending curl requests for the core APIs with SSL. Use the following curl request to test for core APIs with SSL.
+
+
+```
+curl -k -X POST https://localhost:4900/shutdown
+```
+
+### Authentication
+
+The Data Prepper core APIs support HTTP basic authentication. You can set the username and password with the following configuration in the `data-prepper-config.yaml` file:
+
+```yaml
+authentication:
+  http_basic:
+    username: "myuser"
+    password: "mys3cr3t"
+```
+
+You can disable authentication of core endpoints using the following configuration. Use this with caution because the shutdown API and others will be accessible to anybody with network access to your Data Prepper instance.
+
+```yaml
+authentication:
+  unauthenticated:
+```
+
+### Peer Forwarder
+
+Peer Forwarder can be configured to enable stateful aggregation across multiple Data Prepper nodes. For more information about configuring Peer Forwarder, see [Peer Forwarder Configuration](https://github.com/opensearch-project/data-prepper/blob/main/docs/peer_forwarder.md). It is supported by the `service_map_stateful`, `otel_trace_raw`, and `aggregate` processors.
+
+### Shutdown timeouts
+
+When the Data Prepper `shutdown` API is invoked, the `ExecutorService` sink and processor are given time to gracefully shut down and clear any in-flight data. The default graceful shutdown timeout for the `ExecutorService` process is 10 seconds. You can configure the timeout with the following optional YAML file parameters:
```

Comment:
"You can configure the timeout with the following optional YAML file parameters:"
Just to be sure, this configuration is done in the `data-prepper-config.yaml` file. Yes?

Reply:
@oeyh Can you confirm that the YAML file referred to here is the `data-prepper-config.yaml`file?

Reply:
Yes, it is.